### PR TITLE
Modernize GitHub templates (issues and PRs)

### PR DESCRIPTION
**Pull request by bigmontz**

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    See https://neo4j.com/docs/javascript-manual/current/client-applications/#js-driver-logging on
+    how to enable them.
+    Also, **full** stacktraces are very helpful if Exceptions are involved.
+    A minimal reproducer if possible
+    https://stackoverflow.com/help/minimal-reproducible-example
+-->
+
+
+# My Environment
+Node Version: <!-- output of `node --version` -->
```
Comment (robsdedude) [inline .github/ISSUE_TEMPLATE/bug.md]: What about deno :eyes: Or browser?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+-->
+
+<!--
+    Title:
+    PRs targeting the current default branch (nightly driver) don't follow a
+    fixed naming scheme.
+    If your PR is a backport, please use the following naming scheme:
+    [targeted version] description
+    e.g.,
+    [4.4] Remove a banana from the code
```
Comment (robsdedude) [inline .github/PULL_REQUEST_TEMPLATE.md]: This seems to not match the workflow. On the other side this repo seems to use the `cherry-pick` label (which external contributers might not be able to set? not sure),  which is not a thing in the Python repo. So maybe you want to do something with that.

File: .github/ISSUE_TEMPLATE/bug.md (hunk 1 above)
Comment (bigmontz) [inline .github/ISSUE_TEMPLATE/bug.md]:
```suggestion
Javascript Runtime Version: <!-- for instance, the output of `node --version` or the browser version -->
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (bigmontz) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
    If your PR is a backport, please link the original PR in the comments.
```

**Review by robsdedude — APPROVED** [review]
:raccoon:

**Pull request merged**